src/web-ui/src/state/auth/auth.effects.ts

```diff
@@ -0,0 +1,121 @@
+import { Injectable } from '@angular/core';
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import { catchError, of, switchMap, tap } from 'rxjs';
+import { map, mergeMap } from 'rxjs/operators';
+import { AuthenticationApiService } from '../../services/api/authentication-api.service';
+import {
+  getMyself,
+  logout,
+  removeToken,
+  saveToken,
+  setAuthorities,
+  setUser,
+  setUserOnAppInit,
+  signIn
+} from './auth.action';
+import { ToastService, ToastType } from '../../services/toast.service';
+import { AuthenticationRequest, AuthenticationService } from '../../services/authentication.service';
+import { HttpErrorResponse } from '@angular/common/http';
+import { goRoute, showToast } from '../common/common.action';
+import * as _ from 'lodash';
+
+@Injectable()
+export class AuthEffects {
+
+  signInUser$ = createEffect(() =>
```

Comment:
Wygląda na coś skomplikowanego i nie czuję benefitów. Przypomina reacta. Opiszesz co jest lepszego w takim podejściu?

Reply:
Nie powiedziałbym ze skomplikowane. Inicjujemy na start state, wywołujemy jedna akcje z poziomu komponentu, akcja ta jest przechwytywana w efekcie i w tym efekcie odpalamy kolejne akcje i mamy ładnie wydzieloną logikę. Oczywiście będziemy to stosować głównie do autoryzacji, bo dla wszystkich restów jest to kosztowna zmiana, ale wydaje mi się ze w zakresie autoryzacji będzie to miało swoje uzyski.